Added token accuracy checks to `simple_text_demo.py` to replace `test_accuracy.py` in TT-Transformer

epam-ioannis-alexiou opened the pull request:
### Ticket
https://github.com/tenstorrent/tt-metal/issues/21175

### Problem description
simple_text_demo.py does not have accuracy metric included in the script. The aim is to transfer the token accuracy logic from test_accuracy script to the simple_text_demo scirpt while LLM speed performance is evaluated with minimal intervention.

### What's changed

- [x] add the TokenAccuracy class with example test case
  - A class has been added that measures token accuracy based on precomputed tokens from a list of LLM models that can be found at https://github.com/tenstorrent/tt-metal/tree/main/models/tt_transformers/tests/reference_outputs.
  - The call of token accuracy is done by setting True or False the argument token_accuracy in the main function.
The method of the  token accuracy class is activated by if statements in the main script and prints a log of the top1 and top5 accuracy metric.

- [x] add reference input files (such that `reference_data_file = f"models/tt_transformers/tests/reference_outputs/{model_name}.refpt"` can find the correct `.refpt`) for all the ML models that are currently covered by `test_accuracy.py` and demonstrate success in CI pipelines
- [x] ~~delete `test_accuracy.py` and all references to it~~ --> this will be worked on in a separate PR --> issue #23839 to track progress right now
- [x] ~~enable token_accuracy checks on all existing test cases of `simple_text_model.py`~~ --> issue #23839 to track progress right now



### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

yieldthought reviewed (commented):
Interesting! I like the cleanliness of the integration. However, _always_ calling TokenAccuracy and having it automatically turn itself on or off with self.flag feels unexpected and kinda wrong.

I see how wrapping the generator in top1n5_logits makes sense so the sampling can be turned off when running an accuracy test. I don't know, I don't like adding extra layers of indirection between the demo code and the generator/model, each one makes it harder for a user to read the demo and see how to implement their own classes. Obviously by packing more and more testing functionality into the demo we are directly causing stress along this fracture point and this is an example of that, but if there's a smart or creative way to get generator back out of here and make this class an optional part of the calling class rather than a wrapper that is executed in every case that'd feel a lot better.

Maybe we need no more than:
1. Demo optionally provided with an expected reference output for each prompt (e.g. optional json attribute in the existing prompts file)
2. Prompt json file automatically decompressed if .json.gz provided (this is a cute little hack around GitHub size limits, but honestly we could also just take a shorter section of the reference text if we don't want this complexity)
3. Add functionality to Generator to optionally also return the top-5 tokens even when sampling is used and enable this functionality when we have a reference output for the prompt
4. Make this accuracy class literally just worry about tokenizing the reference output and doing the top-1 / top-5 stats when given each set of top-5 token logits, call this when we have a reference output from the prompt

Something like that?

epam-ioannis-alexiou commented:
Bearing in mind your comments:
1. I have removed the text I was loading by using test_accuracy example. Now when accuracy is called I have a script slitting the input by the first encountered question mark and keeping the rest as reference. It made sense to me by reading these prompts.

2.  I have put if statements where calling the class methods in the main demo script as I cannot run all the steps needed for accuracy class to generate the metrics without me duplicating pieces of code. I put the class accuracy methods where I needed to use outputs/methods of the main demo code once.

3. In the generator.py in decode_forward_text I pulled out and put as method argument (argmax_on_device). Also in is_token I replaced it sampling_params boolean check with this argmax_on_device. I see in the code that this is what swaps tokens with logits. Since temperature is not implemented yet and that is dealt in the main simple_text_demo file with sample_host, it would be nice to be left visible in the demo script so any user can modify//implement easily his/hers own temperature function together with softmax as in sample_host.

epam-ioannis-alexiou commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -27,6 +27,45 @@
+        self.store_predicted_tokens = []
+        self.store_predicted_top5_tokens = []
+
+        index = input_prompts[0].find("?")
```
I am see a similar logic when text (not precomputed tokens) is used in text accuracy. In test accuracy script file, part of the text is used as input while the rest is kept as reference to compare against the predicted text from the network. In the above case I stop the input prompt to the first question while keeping the answer separate (I see the answer kept in the same string taken from this input_data_questions_prefill_128.json). My goal is to use the answer as reference. My question  here is what if the user wants to benchmark using this code?  then I need to have data_loaders for reference and generated answers. Let me know how I should design that but looking at the imput examples I thought that should be a way without modifying the code too much.

epam-ioannis-alexiou commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -625,11 +679,13 @@ def test_demo_text(
                 page_table=page_table,
                 kv_cache=tt_kv_cache,
                 sampling_params=device_sampling_params,
+                argmax_on_device=False,
```
What lead me to pull it on arguments and default to False is the need of top5 tokens from network predictions/logits. The decode_forward_text of generator retruns only the top1 token without logits. I saw in the code that argmax_on_device is an important variable but I want logits in order to compute the top5. The reason is that top5 metric needs 5 candidate tokens that are probable to be the next predicted token.

yieldthought reviewed (changes requested):
Does this code still implement teacher forcing? Does it _work_?

It looks to me as if generator will still perform auto-regressive decoding and the only place the reference token is used is for the prompt?

epam-ioannis-alexiou commented:
> Does this code still implement teacher forcing? Does it _work_?
> 
> It looks to me as if generator will still perform auto-regressive decoding and the only place the reference token is used is for the prompt?

Apologies I have added teacher forcing to increase token accuracy. Generally this method is used for training but I undestand the logic since we look for token exact mathing. I have added token_accuracy argument in the main function call and defaulted it to True. Let me know what is the prefered way to have this because I wanted to avoid adding another parameter to pytest.

 I evaluate now on larger sequence (from 128 to 200 tokens) than test_accuracy.py so you may notice a 2% drop roughly. Currently I notice a strange behaviour of the token generation when I switch from performance to accuracy and a drop in token accuracy metric.

gwangTT reviewed (changes requested):
Can you add meaningful contents to the description of this PR, specially "What's Changed" and "Problem Description" sections?

gwangTT reviewed (approved):
Thanks for adding the PR description! 

The PR looks good to me except for a point that needs clarification. 👇 

Approved and let's get CI tests pased before merging

skhorasganiTT commented on models/tt_transformers/tt/generator.py:
```diff
@@ -208,12 +208,13 @@ def decode_forward_text(
         read_from_device=True,
-        sampling_params: SamplingParams = None,  # Should be None if not greedy decoding / sampling on device.
+        sampling_params: SamplingParams = None,
+        argmax_on_device=False,  # Should be None if not greedy decoding / sampling on device.
```
An extra `argmax_on_device` param is redundant, device sampling params are already contained in `sampling_params` and argmax_on_device is inferred based on the statement below `argmax_on_device = sampling_params is not None and sampling_params.temperature == 0`.

gwangTT commented:
@epam-ioannis-alexiou, are you running CI tests on this? I suggest running: 
[All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes,
[Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
and https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml

epam-ioannis-alexiou commented:
https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml?query=branch%3Aioannisalx-21175_accuracy_test_mode

https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml?query=branch%3Aioannisalx-21175_accuracy_test_mode

https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml?query=branch%3Aioannisalx-21175_accuracy_test_mode

@gwangTT  please have a look and let me know

gwangTT commented:
Thanks, Ioannis! The failed test cases in the linked runs seem unrelated to this PR.

I caught up with Ioannis offline and we decided that test cases with `token_accuracy=True` should be added to demonstrate the use of the work in this PR.

epam-ioannis-alexiou commented:
I have added another CI case to test token accuracy and changed the token accuracy class constructor to search for the model name inside the directory where the refpt files reside.

mo-tenstorrent commented on tt_metal/impl/profiler/profiler.hpp:
```diff
@@ -281,6 +281,12 @@ class DeviceProfiler {
 
     // Read data from profiler buffer using slow dispatch
     void issueSlowDispatchReadFromProfilerBuffer(IDevice* device);
+
```
I believe changes here and on the cpp file are rebase conflicts not real changes.

epam-ioannis-alexiou replied:
Yes I have not touched this file. Let me know if there is anything I can do from part to resolve this.

skhorasganiTT commented on models/tt_transformers/tt/generator.py:
```diff
@@ -205,12 +205,13 @@ def decode_forward_text(
         enable_trace=True,
         read_from_device=True,
-        sampling_params: SamplingParams = None,  # Should be None if not greedy decoding / sampling on device.
+        sampling_params: SamplingParams = None,
+        argmax_on_device=False,  # Should be None if not greedy decoding / sampling on device.
```
rebase conflicts

epam-ioannis-alexiou replied:
I definetly do not need this change I want generator.py unchanged as in main branch

gwangTT reviewed (commented):
The change in [99cf929](https://github.com/tenstorrent/tt-metal/pull/22317/commits/99cf92992a0f25acfa9dd446517e792399153f6e) does not accomplish much new functionality. I think the old code in that diff should work just fine if the correct model_name is passed in. 

The only thing that is left for this PR to do is to confirm that all the models that requires token accuracy tests can be covered by the code change in this PR through `simple_text_demo.py`. For that, CI tests of `single card (demo)` (https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml), `T3000 Demo Test` (https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) need to be run and the results of the token accuracy checks need to be verified. @epam-ioannis-alexiou, please click into the Actions page, kick off those runs manually (like I briefly showed in our meeting but let me know if you want to meet again about this), and paste links to those runs to the comments of this PR.

I also noticed that there are a lot of rebase conflicts (white space changes, imports ordering, etc.) with files that are not part of this PR. You can revert out of those conflicts.

gwangTT commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -48,7 +48,7 @@ def prepare_ref_tokens(self, tokenizer):
         self.store_predicted_tokens.append(tokens)
         self.gt_pos += 1
-        return self.gt_tokens[self.gt_pos].unsqueeze(-1).unsqueeze(-1)
+        return self.gt_tokens[min(self.gt_pos,len(self.gt_tokens))].unsqueeze(-1).unsqueeze(-1)
```
I am a little confused by this change as the old code makes more sense to me. Does the user of `collect_predicted_tokens` change its behavior?

gwangTT reviewed (changes requested):
Thanks, @epam-ioannis-alexiou! Please address the comments I just added. 

I can see the CI tests printing out token accuracy tests. Very cool!

gwangTT commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -761,7 +762,7 @@ def test_demo_text(
 
             if token_accuracy:
-                out_tok = token_acc.collect_predicted_tokens(out_tok.item())
+                out_tok[0] = token_acc.collect_predicted_tokens(out_tok[0].item())
```
I think the right approach to follow `test_accuracy.py`:
- load your input tokens and expected top-5 tokens from the .refpt file
- send the input tokens as the prompt to the LLM model and get the out_toks
- compare the out_tok against the expected top-5 tokens

BTW, you do not need to enable token_accuracy for every test case in `simple_text_demo.py`. Can focus on adding the same test cases that matches what `test_accuracy.py` is testing.

gwangTT replied:
Resolved during offline discussion -- `test_accuracy.py` is also following the approach that this PR is already implementing, i.e., resetting `out_tok` to the reference tensor value for the next round of decoding.

@epam-ioannis-alexiou is going to add comment to explain this approach.

epam-ioannis-alexiou replied:
Ok no prob I will update the code to report accuracy for batch_size = 1
Below I post the test of T3000 and sing card:
T3000 -> https://github.com/tenstorrent/tt-metal/actions/runs/15968363590/job/45034525720
and single card -> https://github.com/tenstorrent/tt-metal/actions/runs/15906035749/job/44862013454

gwangTT reviewed (approved):
LGTM with one small change of adding comment on the use of reference tensor to reset `out_tok`.

Thanks, @epam-ioannis-alexiou!

mtairum commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -249,6 +293,7 @@ def prepare_generator_args(
             True,  # stop_at_eos
             False,  # ci_only
             1,  # data_parallel
+            True,  # token_accuracy
```
Shouldn't this be False?

mtairum commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -266,6 +311,7 @@ def prepare_generator_args(
             False,  # stop_at_eos
             False,  # ci_only
             1,  # data_parallel
+            True,  # token_accuracy
```
Shouldn't this be False?

mtairum commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -280,6 +326,7 @@ def prepare_generator_args(
             False,  # stop_at_eos
             True,  # ci_only
             1,  # data_parallel
+            True,  # token_accuracy
```
Shouldn't this be False?

mtairum commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -308,6 +356,7 @@ def prepare_generator_args(
             True,  # stop_at_eos
             False,  # ci_only
             4,  # data_parallel
+            True,  # token_accuracy
```
Shouldn't this be False?

mtairum commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -350,6 +401,7 @@ def prepare_generator_args(
             False,  # stop_at_eos
             True,  # ci_only
             4,  # data_parallel
+            True,  # token_accuracy
```
Shouldn't this be False?

mtairum commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -392,6 +446,7 @@ def prepare_generator_args(
             True,  # stop_at_eos
             True,  # ci_only
             32,  # data_parallel
+            True,  # token_accuracy
```
Shouldn't this be False?

mtairum reviewed (changes requested):
Can you also add an override for the new token_accuracy option?

This should be added to the main demo script (around line 570):
```
token_accuracy = request.config.getoption("--token_accuracy") or token_accuracy
```

And the option should be added to `models/tt_transformers/demo/conftest.py`

mtairum reviewed (changes requested):
Please also update `models/tt_transformers/README.md`. 

References to the old accuracy test should be removed. And a passage explaining how to run a teacher-forcing accuracy test from the `simple_text_demo.py` should be added.

mtairum commented:
The PR seems good to me. Please address the minor requests made above, rebase and run the tests below (and link the runs here) -> then we can merge this.

Given the changes don't touch existent tests I don't think you'll need to run all the tests. 

But please rebase your branch to the latest main and then run the following tests:
- All-post commit (should always be run on the final version of PR): https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml
- single-device demo: https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml (all good if llama3 tests pass, ignore other models).

epam-ioannis-alexiou commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -193,6 +233,7 @@ def prepare_generator_args(
             True,  # stop_at_eos
             False,  # ci_only
             1,
+            True,  # token_accuracy
```
I need to have one test case set to True to see the accuracy results. I can have this one and set the rest below to False. 
Shall I do that ?

epam-ioannis-alexiou commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -580,8 +653,16 @@ def test_demo_text(
+
     num_tokens_generated_decode = []
 
+    # cannot change max_generated_tokens
```
It is from the previous author. It refers to the code next below. I have understood that if the inference start the number of max__generated_tokens cannot modified during the token generation phase.

mtairum commented:
@epam-ioannis-alexiou This rebase clearly has gone wrong. Please backup your local branch (just to make sure you don't lose any work) and either force push a proper rebased branch, or create a new PR.

epam-ioannis-alexiou committed:
add accuracy test mode to simple_text_demo

mtairum commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -167,7 +207,7 @@ def prepare_generator_args(
 # instruct (bool): Whether to use instruct weights or general weights
 # repeat_batches (int): Number of consecutive batches of users to run (default: 1)
-# max_seq_len (int): Maximum context length supported by the model (Llama-3.1 and Llama-3.2 models have a maximum context length of 128k, i.e., 128 * 1024)
+# max_seq_len (int): Maximum context length supported by the model (Llama3.1 and Llama3.2 models have a maximum context length of 128k, i.e., 128 * 1024)
```
These should have the hyphen (-)

mtairum commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -458,11 +530,13 @@ def test_demo_text(
     optimizations,
     stop_at_eos,
     mesh_device,
+    use_program_cache,
```
Remove (this is a remnant of the rebase)

mtairum commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -579,8 +654,16 @@ def test_demo_text(
 
     generator = Generator(model, model_args, mesh_device, tokenizer=tokenizer)
 
+    if token_accuracy:
```
this `if token_accuracy` statement can be merged with the one above.

mtairum replied:
And put a comment mentioning we're doing teacher forcing here.

mtairum commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -671,12 +754,16 @@ def test_demo_text(
         logger.info(f"Starting decode loop...")
 
         # Log total inference (accounting for compile_decode as well)
+        store_generated_tokens = []
```
This is not used anywhere?

epam-ioannis-alexiou replied:
I removed it thanks

mtairum commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -671,12 +754,16 @@ def test_demo_text(
                 profiler.start(f"compile_decode", iteration=batch_idx)
             else:
                 profiler.start(f"inference_decode_time_{iteration}", iteration=batch_idx)
+            # below the collect method also applies teacher forcing which necessary for exact token matching
```
Missing a `is` in `which is necessary`

mtairum commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -847,22 +938,23 @@ def test_demo_text(
 
     # Benchmark targets
-    supported_models = ["Llama-3.2-1B", "Llama-3.2-3B", "Llama-3.1-8B", "Llama-3.2-11B", "Llama-3.1-70B", "Mistral-7B"]
+    supported_models = ["Llama3.2-1B", "Llama3.2-3B", "Llama3.1-8B", "Llama3.2-11B", "Llama3.1-70B", "Mistral-7B"]
```
These should not be changed.

epam-ioannis-alexiou replied:
In the main branch the hyphen was added back in, Mine passed the tests without it last week. I will try to incorporate this change and test again

mtairum commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -847,22 +938,23 @@ def test_demo_text(
     if model_name in supported_models:
         assert tt_device_name in supported_devices, f"Device {tt_device_name} not supported"
 
+        model_device_key = f"{tt_device_name}_{model_name}"
```
Why was this moved? (rebase artifact?)

epam-ioannis-alexiou replied:
I have moved up as in main. I had an old version from main

mtairum commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -873,30 +965,30 @@ def test_demo_text(
+            "N150_Llama3.2-1B": 160,
+            "N300_Llama3.2-1B": 250,  # TODO Update target
+            "T3K_Llama3.2-1B": 300,  # TODO Update target
+            "TG_Llama3.2-1B": 300,  # TODO Update target
             #
-            "N150_Llama-3.2-3B": 60,
-            "N300_Llama-3.2-3B": 100,  # TODO Update target
-            "T3K_Llama-3.2-3B": 150,  # TODO Update target
-            "TG_Llama-3.2-3B": 150,  # TODO Update target
+            "N150_Llama3.2-3B": 60,
+            "N300_Llama3.2-3B": 100,  # TODO Update target
+            "T3K_Llama3.2-3B": 150,  # TODO Update target
+            "TG_Llama3.2-3B": 150,  # TODO Update target
             #
-            "N150_Llama-3.1-8B": 23,
-            "P150_Llama-3.1-8B": 23,  # TODO Update target
-            "N300_Llama-3.1-8B": 38,
-            "P300_Llama-3.1-8B": 38,
-            "T3K_Llama-3.1-8B": 45,
-            "TG_Llama-3.1-8B": 45,  # TODO Update target
+            "N150_Llama3.1-8B": 23,
+            "P150_Llama3.1-8B": 23,  # TODO Update target
+            "N300_Llama3.1-8B": 38,
+            "P300_Llama3.1-8B": 38,
+            "T3K_Llama3.1-8B": 45,
+            "TG_Llama3.1-8B": 45,  # TODO Update target
             #
-            "N150_Llama-3.2-11B": 23,
-            "N300_Llama-3.2-11B": 38,  # TODO Update target
-            "T3K_Llama-3.2-11B": 45,  # TODO Update target
-            "TG_Llama-3.2-11B": 45,  # TODO Update target
+            "N150_Llama3.2-11B": 23,
+            "N300_Llama3.2-11B": 38,  # TODO Update target
+            "T3K_Llama3.2-11B": 45,  # TODO Update target
+            "TG_Llama3.2-11B": 45,  # TODO Update target
             #
-            "T3K_Llama-3.1-70B": 20,  # TODO Update target
-            "TG_Llama-3.1-70B": 20,  # TODO Update target
+            "T3K_Llama3.1-70B": 20,  # TODO Update target
+            "TG_Llama3.1-70B": 20,  # TODO Update target
```
Please put the hyphen (-) back on these. I'm assuming it was a rebase issue.

epam-ioannis-alexiou commented on models/tt_transformers/demo/simple_text_demo.py:
```diff
@@ -962,58 +1053,3 @@ def test_demo_text(
-                "N300_Qwen2.5-7B": 150,
-                # T3K targets
-                "T3K_Llama3.1-70B": 181,
-                "T3K_Qwen2.5-72B": 211,
-                "T3K_Qwen3-32B": 470,
-            }
-            ci_target_decode_tok_s_u = {
-                # N150 targets - higher is better
-                "N150_Llama3.2-1B": 58,
-                "N150_Llama3.2-3B": 35,
-                "N150_Llama3.1-8B": 21,
-                "N150_Mistral-7B": 23,
-                # N300 targets
-                "N300_Qwen2.5-7B": 10,
-                # T3K targets
-                "T3K_Llama3.1-70B": 14,
-                "T3K_Qwen2.5-72B": 13,
-                "T3K_Qwen3-32B": 11,
-            }
-
-            # Only call verify_perf if the model_device_key exists in the targets
-            ci_targets = {}
-            if model_device_key in ci_target_ttft:
-                ci_targets["prefill_time_to_token"] = ci_target_ttft[model_device_key] / 1000  # convert to seconds
-            if model_device_key in ci_target_decode_tok_s_u:
-                ci_targets["decode_t/s/u"] = ci_target_decode_tok_s_u[model_device_key]
-                # calculate from per-user rate
-                ci_targets["decode_t/s"] = ci_target_decode_tok_s_u[model_device_key] * global_batch_size
-
-            if ci_targets:  # Only verify performance if we have targets for this model/device combination
-                verify_perf(
-                    measurements,
-                    ci_targets,
-                    high_tol_percentage=1.15,
-                    expected_measurements={k: True for k in ci_targets.keys()},
-                )
-            else:
-                logger.warning(
-                    f"No CI performance targets found for {model_device_key}. Skipping performance verification."
-                )
```
caused because of rebase. This piece of code is not relevant anymore and it is neither found on this branch nor main branch. It is outdate block of code eventhough the ci_targets parameters are still in the code

epam-ioannis-alexiou committed:
Merge branch 'main' into ioannisalx-21175_accuracy_test_mode

epam-ioannis-alexiou committed:
fixed typo conftest.py

epam-ioannis-alexiou committed:
second fix conftest.py

epam-ioannis-alexiou committed:
incorporated changes from main and addressed comments simple_text_demo.py

epam-ioannis-alexiou committed:
code comment fixed  simple_text_demo.py

epam-ioannis-alexiou committed:
Merge branch 'main' into ioannisalx-21175_accuracy_test_mode

epam-ioannis-alexiou commented:
Singles -> https://github.com/tenstorrent/tt-metal/actions/runs/16170223217
T3000 -> https://github.com/tenstorrent/tt-metal/actions/runs/16170231972

epam-ioannis-alexiou committed:
Merge branch 'main' into ioannisalx-21175_accuracy_test_mode

epam-ioannis-alexiou committed:
Merge branch 'main' into ioannisalx-21175_accuracy_test_mode

epam-ioannis-alexiou committed:
Merge branch 'main' into ioannisalx-21175_accuracy_test_mode

epam-ioannis-alexiou committed:
Merge branch 'main' into ioannisalx-21175_accuracy_test_mode

epam-ioannis-alexiou committed:
Merge branch 'main' into ioannisalx-21175_accuracy_test_mode

epam-ioannis-alexiou committed:
Merge branch 'main' into ioannisalx-21175_accuracy_test_mode
